content/terraform/v1.12.x/docs/language/values/variables.mdx

```diff
@@ -1,29 +1,19 @@
 ---
-page_title: Set configuration parameters
-description: Learn how to make Terraform modules flexible, composable, and reusable by defining input variables, locally-scoped values, and output values for your module.
+page_title: Use variables to input module arguments
+description: Learn how to variables define the interface of your module, and how to assign values to variables.
 ---
 
-# Set configuration parameters
-
-Make your Terraform modules flexible, composable, and reusable by defining your module's input variables, defining module-scoped local values, and defining output values to expose data from your module.
-
-## Overview
-
-Use the following blocks to modify your module's behavior:
-
-- The `variable` block lets you define input arguments, letting module consumers pass in values to customize behavior at runtime.
-- The `locals` block defines temporary values scoped to a module, letting you name and reuse expressions in your configuration.
-- The `outputs` block exposes data from a module, making module results available in the Terraform CLI, HCP Terraform, and other parts of your configuration.
-
-## Define module input arguments with variables
+# Use variables to input module arguments
 
 > **Hands-on:** Try the [Customize Terraform Configuration with Variables](/terraform/tutorials/configuration-language/variables?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial.
 
-Add `variable` blocks to your configuration so that module consumers can pass in specific input values. This lets module consumers customize module behavior without altering the module's source code. Variables define the interface of your module by specifying what values your module accepts as arguments.
+Variables define the interface of your module by specifying what values your module accepts as arguments. The `variable` block lets your module consumers customize module behavior without altering the module's source code.
```

Comment:
I would consider removing lines 10-12 and starting with the paragraph at line 35 following by the paragraph at line 14.

Reply:
I'd prefer the intro was broad, so folks understand what this page is about, before it gets _super_ specific in the following sections. 

I'll split the difference with:

"
Variables define the input interface of your module by specifying what values your module accepts as arguments.

Terraform sets up resources with hardcoded values the same way every time, making your module inflexible and potentially hard to reuse. If you know a value in your configuration changes between Terraform operations, you can replace that value with a `variable` block. A `variable` block lets module consumers customize module behavior without altering the module's source code.
"

then leaving the following section so they can learn how to define a `variable` block.